src/api/apiService.js

```diff
@@ -15,53 +15,47 @@ const saveSessionHeaders = (headers) => {
   }
 };
 
-const handleErrors = response =>
+const handleErrors = (response, requestMethod) =>
   new Promise((resolve, reject) => {
+    const { status, ok, headers, statusText } = response;
     if (!response) {
       reject(new Error({ message: 'No response returned from fetch' }));
       return;
     }
 
-    saveSessionHeaders(response.headers);
-    if (response.ok) {
-      resolve(response);
+    saveSessionHeaders(headers);
+    if (ok && (status === 200 || status === 204)) {
+      if (requestMethod === 'delete') {
+        resolve();
+        return;
+      }
+
+      resolve(response.json());
```

Comment:
You are changing the logic when the status code is `204`. That code means no content in the body and that's why before it was doing:
```
  const bodyIsEmpty = response.status === 204;
  if (bodyIsEmpty) {
    return Promise.resolve();
  }
```
If instead of resolving the promise in that case you call `response.json()` you should be getting an error because the body of the response is empty and you cannot do `.json()`

Reply:
```
if (ok && (status === 200 || status === 204)) {
      if (requestMethod === 'delete') {
        resolve();
        return;
      }

      resolve(response.json());
      return;
    }
```
The status code 204 should be only to Delete or Put method.

Reply:
https://developer.mozilla.org/es/docs/Web/HTTP/Status/200

Reply:
There could be cases I think where it's a `POST` and the backend still does not return any content because it's not necessary. I think in my project I have one of those cases.

The link you sent says that more often than not `DELETE` and `PUT` return `204` instead of `200` but it does not say that a `POST` can't return a `204`

Reply:
In any case, if it's a `204` you shouldn't be doing `response.json()` no matter the requestMethod because it will yield an error.